src/Aspire.Dashboard/Configuration/ValidateDashboardOptions.cs

```diff
@@ -0,0 +1,45 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.Extensions.Options;
+
+namespace Aspire.Dashboard.Configuration;
+
+public sealed class ValidateDashboardOptions : IValidateOptions<DashboardOptions>
+{
+    public ValidateOptionsResult Validate(string? name, DashboardOptions options)
```

Comment:
Check for valid uris in here?

Reply:
I made parsing part of setting the properties. It will get validated when the value is set (error if it fails).

Reply:
Just beware of this https://github.com/dotnet/runtime/issues/44381

Reply:
Hmmm, ok! I'll change it.

Reply:
Done. If I had more time, I'd look at what other places do here, but I think this is functional.